[Merged by Bors] - Clean up log output

lrettig opened the pull request:
## Motivation
Closes #1570
Closes #1923
Closes #1973

## Changes
One minor bugfix (489089398e1170232cf89947ae9a038458d98994). The rest of the changes are cosmetic, affecting log output only.

## TODO

- [x] Fix bytes representations to something readable: `Failed sending HARE_PROTOCOL msg [123 129 185 230 148 211 148 26 48 31 95 93] to` (already fixed in 1fc34b2a9e2578570b33077cb1371e6f8d743420)
- [x] In the message `Failed sending newBlock msg [22 91 102 90 183 144 70 218 165 121 166 12] to 6S3sSsvzYv4kb2oQLsbCHae2xS5GtLKU6JvTCMMfG6uP, reason=connection was closed` the peer name is not the correct peer we failed sending the message to.
- [x] In the log could not get result for layer in range there is "error":"no result for the requested layer" The word error should not appear (#1951)
- [x] Block id is now a short string/id of 5 characters -- causes too collisions (in the logs) -- should be longer (12 char. ?)
- [x] "got new ATX" should include all possible information: similar to the info in "ATX published!" + other ATX fields (in particular `epoch_id').
- [ ] fix JSON in starting build atx in epoch %v and re-entering atx creation in epoch %v (**CAN'T FIND**)
- [x] late hare messages (late preround / "contextually invalid message") should include the node_id that sent the message. "contextually invalid" is not a good name - should be "late" / "not in time" (if too early)
- [x] we have `sending message failed %!(EXTRA *service.DataMsgWrapper=&{true 6 1 [113 169 217 46 19 216 93 89 35 103 13 132 9 97 7 74 149 179 217 161 81 4 131 209 229 113 105 88 151 40 90 48]}, string= error: , *errors.errorString=connection was closed)` as well as `peer responded with nil to poet request DFjCnmocNHoa3Kyd16LWQpFTTcRZThP1CS6vuB8WvFEj%!(EXTRA []uint8=[113 169 217 46 19 216 93 89 35 103 13 132 9 97 7 74 149 179 217 161 81 4 131 209 229 113 105 88 151 40 90 48])` should fix the presentation to something comprehensible (in all relevant places)
- [x] see also:
`handle atx request {atx_ids %!s(zapcore.FieldType=15) %!s(int64=0) [ca1cfd21703c8ed26655722c3da2f37a009ec7aeabe2131140498422af57514a b1a56aa5afc2d523f7d1c907c88bd579ecaa29d5dbfaf947cbc9d06d84c19a67 d5b4af7d28cf9df9d8ef568e843a7544cd5a78d5a1760ee91971441d851a61c9 789885c3ac2307cad568ca9fa70de707dcb434b8353fe923f396cd1180ba8647 6c34150b8a2bd843246185c3b8f84a2f15424493ce4608c0ed25426db941026a ] <nil>}`
- [x] Reward calculated appears twice in each layer
- [x] Fix `atx size %!s(int=18956)%!(EXTRA log.Field={atx_id 15 0 f6f9c <nil>})`. It is better if this is part of the "ATX published!" log
- [x] Add epoch_id to "I've created a block"
- [x] `unfamiliar atx was requested (id: %s)`, fix %s
- [x] Add duration to "finished PoST execution"
- [x] Any log about the initial post creation / proof root etc.?
- [x] the function CalcActiveSetSize should print the calculated active set size. Can write it under done calculating active set size
- [x] `block created` should have associated ATX in short string
- [x] `fetched all block data` or `adding block` should have associated ATX in short string
- [ ] already synced for layer appears every 10 seconds. Consider the benefit of printing it (**CAN'T FIND**; `node is synced`?)
- [ ] "release tick" log statement should include epoch ID and node ID (#1980)
- [x] Blocks coming by fetching (as opposed to gossip) should have all the logs as in got new block. Currently some logs are under the start handle block log (in the handleBlock() function).
- [x] got new atx should have longer ids than 5 characters (atx_id, _sender_id etc.) to prevent collisions and displaying inaccurate information. (For dashboard purposes)
 - [x] `miner %v is eligible for %d blocks on %d layers in epoch %d` should print the actual layers not just no. of blocks in this epoch
 - [ ] `new state root` printed twice per layer (won't fix: https://github.com/spacemeshos/go-spacemesh/pull/1952#issuecomment-612502720)
 - [ ] @ilans: can every log line include node_id, epoch, and layer? (#1980)
 - [x] `certificate validation duration` and `SVP validation duration` => DEBUG
 - [x] Hare prints `Should not participate` but it doesn't print anything if the node is participating in this round
 - [x] `2020-03-26T03:28:45.451Z        INFO    21f52.sync              handle tx request {atx_ids %!s(zapcore.FieldType=15) %!s(int64=0) [30783166303036303034653361383433613438653630633031393830366330646531393865...`
- [x] ALL log messages regarding messages from p2p (not local things) should include the sender's node_id.
- [x] `block has no missing blocks in view` INFO -> DEBUG
- [x] `Allocated cache and file handles`

lrettig committed:
Misc minor cleanup

lrettig committed:
Print reward log statement once per layer

Move log up one level and include both per-block and per-layer info in
one log statement.

lrettig committed:
Update test

lrettig committed:
Display block and ATX IDs as longer string

Five characters is not enough, there are too many collisions in the
logs. Expand this to 12. Consolidate the code path that controls the
lengths.

lrettig committed:
Update test to match previous change

Expect a longer ATX ID

lrettig committed:
WIP: Harmonize how ATXs are printed

Create a new ATX method that returns a standard set of log fields

lrettig committed:
Add ATX broadcast to test

lrettig committed:
Finish standardization of ATX logs

- Pass layersPerEpoch into the block builder
- Pass in ATX length (rather than trying to recalculate it)

Everything else is intrinsic to the ATX.

Also update tests.

lrettig committed:
Harmonize Hare message errors

lrettig committed:
Fix typo

lrettig committed:
Test SendRequest error too

lrettig committed:
Clean up error

lrettig committed:
Cleanup handle atx request errors

lrettig committed:
Make failed to publish atx error clearer

s/error/errmsg/ to make it easier to search the logs

lrettig committed:
Improve block output

on new block created

lrettig committed:
Clean up logs

Format things better, use With() wherever possible.
Fix some typos.

lrettig committed:
Clean up log for PoST execution

Print duration

lrettig committed:
Fix lingering issue from merge conflict

lrettig committed:
Log active set size

lrettig committed:
Clean up logging of ATXIDs

Also print atx and tx count, and ATXIDs, after "fetched all block data"

lrettig committed:
Standardize logging of blocks

lrettig committed:
Grammar in comments

lrettig committed:
Minor fixes to log statements

lrettig commented:
> "release tick" log statement should include epoch ID and node ID (why is it missing node ID?)

This one is tough. Neither the ticker nor the clock, which it lives inside of (and which instantiates it), knows anything about epochs or about the current node. I don't know why the log statement isn't printing the node_id - maybe something to do with the fact that it's an "event"?

Will handle separately in #1980, this PR is long enough already

lrettig committed:
Minor log cleanup

lrettig committed:
Move a const to a better place

To avoid circular dependencies: common/types test -> miner ->
common/types

lrettig committed:
Fix a segfault in block minerID

Add a test for the same

lrettig committed:
Improve block logging in handleBlock

lrettig commented:
>  new state root printed twice per layer

Once is after applying rewards, the other time is after applying transactions. If there are no transactions, the state root will be the same. I think this behavior is correct so I won't change this.

lrettig committed:
Move noisy log messages from Info to Debug

lrettig committed:
Improve hare participate logs

Log when we should participate as well as when we shouldn't

lrettig committed:
Improve logging of layer eligibility

Pretty-print the number of eligible blocks per layer for each epoch

lrettig committed:
Cleanup string handling

lrettig committed:
Minor log cleanup

lrettig committed:
Cleanup tx_ids logging

lrettig committed:
Logline INFO -> DEBUG

don't think we need to see this in INFO level logs

lrettig committed:
Make log statements more readable

Whitespace, newlines

lrettig committed:
Fix typos

lrettig commented:
@ilans please review to make sure you're aware of these changes!
@noamnelke would you mind reviewing? 🙏🙏🙏

noamnelke commented on activation/nipst.go:
```diff
@@ -214,18 +215,19 @@ func (nb *NIPSTBuilder) BuildNIPST(challenge *types.Hash32, atxExpired, stop cha
 			return nil, fmt.Errorf("failed to execute PoST: %v", err)
 		}
 
-		//todo remove (#1923)
-		//this is just for debugging
 		b, _ := types.InterfaceToBytes(proof)
```
We serialize the proof here just to print the length of the serialized string. This todo was to remove the extra serialization for performance reasons.

Three options:
1. Keep the todo
2. Remove the length of the proof from the log and remove the extra serialization (<-- my fav.)
3. Find a way to get this size without doing an extra serialization

I've noticed that you've already closed #1923. This is not good practice. You can link that issue to this PR and it will be closed when you close the PR. This will also add a comment to the issue automatically so people will know to look here before working on it.

lrettig replied:
I made change 2. as you suggested.

I looked into this, and 3. would probably require changing the Proof type to track/cache its own size, or something of this sort, which seems too deep a change for a simple debug statement.

noamnelke commented on common/types/activation.go:
```diff
@@ -173,6 +174,45 @@ func (atx *ActivationTx) InnerBytes() ([]byte, error) {
+
+	}
+
+	return []log.LoggableField{
+		log.AtxID(atx.ShortString()),
+		log.String("sender_id", atx.NodeID.ShortString()),
+		log.String("prev_atx_id", atx.PrevATXID.ShortString()),
+		log.String("pos_atx_id", atx.PositioningATX.ShortString()),
+		log.LayerID(uint64(atx.PubLayerID)),
+		log.EpochID(uint64(atx.PubLayerID.GetEpoch(layersPerEpoch))),
+		log.Uint32("active_set", atx.ActiveSetSize),
+		log.Int("viewlen", len(atx.View)),
+		log.Uint64("sequence_number", atx.Sequence),
+		log.String("NIPSTChallenge", challenge),
+		log.String("commitment", commitmentStr),
+		log.Int("atx_size", size),
+	}
```
Remember my comment about loggable fields?

```suggestion
	return []log.LoggableField{
		atx.ID(),
		log.String("sender_id", atx.NodeID.ShortString()),
		log.String("prev_atx_id", atx.PrevATXID.ShortString()),
		log.String("pos_atx_id", atx.PositioningATX.ShortString()),
		atx.PubLayerID,
		atx.PubLayerID.GetEpoch(layersPerEpoch),
		log.Uint32("active_set", atx.ActiveSetSize),
		log.Int("viewlen", len(atx.View)),
		log.Uint64("sequence_number", atx.Sequence),
		log.String("NIPSTChallenge", challenge),
		log.String("commitment", commitmentStr),
		log.Int("atx_size", size),
	}
```

I want to eventually remove methods like `log.AtxID`, so let's start using the bare form.

noamnelke commented on common/types/block.go:
```diff
@@ -11,16 +11,21 @@ import (
-	return id.AsHash32().ShortString()
+	return id.AsHash32().MediumString()
 }
 
 // Field returns a log field. Implements the LoggableField interface.
-func (id BlockID) Field() log.Field { return log.String("block_id", id.AsHash32().ShortString()) }
+func (id BlockID) Field() log.LoggableField {
```
because you changed the return type from `log.Field` to `log.LoggableField` here, `BlockID` now no longer implements the `log.LoggableField` interface and therefore can't be used as a field directly...

You've made this change in many places and I don't understand the motivation for it...

Worth noting that `log.Field` itself implements the `log.LoggableField` interface, so it should work for most use-cases the way it was.

noamnelke commented on common/types/hashes.go:
```diff
@@ -171,10 +171,22 @@ func (h Hash32) String() string {
 }
 
-// ShortString returns a the first 5 characters of the hash, for logging purposes.
+// MediumString returns the first 10 characters of the hash, for logging purposes.
```
I don't mind having this, but I wonder what the motivation is... Have we seen cases where the 5-char `ShortString` is not enough and there were collisions?

lrettig replied:
Yes, see the original motivation for this PR in #1570 - Barak was running into collisions with block id!

noamnelke replied:
Yeah, so let's make `ShortString()` return a 10-character string instead of 5.

noamnelke reviewed (commented):
This is a big change, so it'll take me a while to review, this is just the first batch of comments. I'll only be able to give it more time next week, sorry about that.

lrettig committed:
Fix database log message

lrettig committed:
Remove spurious serialization

This was only for debug purposes. Remove it as it's unnecessary.
Discussion at
https://github.com/spacemeshos/go-spacemesh/pull/1952#discussion_r409072489.

lrettig committed:
Revert LoggableField -> Field

Make sure these types still implement the `log.LoggableField` interface.

Per @noamnelke's feedback in
https://github.com/spacemeshos/go-spacemesh/pull/1952#discussion_r409080227.

lrettig committed:
Update common/types/activation.go

Take advantage of types that implement LoggableFields per @noamnelke's suggestion

Co-authored-by: Noam Nelke <NoamNelke@gmail.com>

lrettig committed:
Comments

lrettig commented:
@noamnelke I fixed the `Field`/`LoggableField` type issues. Let me know if you're okay with the way I'm assembling an array of `LoggableField`s for data structures such as `Block`:

https://github.com/spacemeshos/go-spacemesh/pull/1952/files#diff-a5963108f18ed56664af656e1536751eR157

also `ActivationTx`:

https://github.com/spacemeshos/go-spacemesh/pull/1952/files#diff-fbc8f466cd0cde2afe76fa37ca7ee514R178

In the case of `Block`, it might be possible to use the same `LoggableField` interface trick you explained to turn the whole thing into a string (so that it could be logged directly), but it would require a bunch of string manipulation. That's why I think returning an array of field types is cleaner.

In the case of `ActivationTx`, outputting a complete description of the data structure also requires passing in parameters (`layersPerEpoch`, `size`) so that's probably not an option.

lrettig committed:
Fix some small typos

noamnelke commented on common/types/activation.go:
```diff
@@ -173,6 +174,45 @@ func (atx *ActivationTx) InnerBytes() ([]byte, error) {
 
+// Fields returns an array of LoggableFields for logging
+func (atx *ActivationTx) Fields(layersPerEpoch uint16, size int) []log.LoggableField {
+	commitmentStr := "nil"
```
Why log the string `nil`? Doesn't it make more sense to only log this field if a commitment is present? Elasticsearch lets you query on field existence, if that's the concern.

Only each miner's first ATX ever should have this field, so it makes more sense to me to only include it when it's relevant.

lrettig replied:
I think I made it log "nil" rather than an empty string because it's more readable to human eyes. I'd prefer to err on the side of keeping the code simpler, especially for logs (i.e., by always logging the same set of fields, not making this conditional). Let me know if you disagree.

noamnelke replied:
I agree that the string "nil" is better than an empty string and I understand your consideration of keeping the code simple. Having said that, I don't think this would complicate the code... You already have a conditional that overwrites the "nil" string.

I don't want to impose my style onto you, so feel free to reject what I suggest if you disagree, but what I had in mind is writing the function like this:
```golang
func (atx *ActivationTx) Fields(layersPerEpoch uint16, size int) []log.LoggableField {
	fields := []log.LoggableField{
		atx.ID(),
		log.String("sender_id", atx.NodeID.ShortString()),
		log.String("prev_atx_id", atx.PrevATXID.ShortString()),
		log.String("pos_atx_id", atx.PositioningATX.ShortString()),
		atx.PubLayerID,
		atx.PubLayerID.GetEpoch(layersPerEpoch),
		log.Uint32("active_set", atx.ActiveSetSize),
		log.Int("view_len", len(atx.View)),
		log.Uint64("sequence_number", atx.Sequence),
		log.Int("atx_size", size),
	}
	if atx.Commitment != nil {
		fields = append(fields, log.String("commitment", atx.Commitment.String()))
	}
	if h, err := atx.NIPSTChallenge.Hash(); err == nil && h != nil {
		fields = append(fields, log.String("nipst_challenge", h.String()))
	}
	return fields
}
```

lrettig replied:
I changed "nil" to the empty string here, so that `commitmentStr` and `challenge` are handled the same way. However, I disagree with leaving the fields out entirely if they're empty. Having spent countless hours of my life looking at logs and hunting down bugs, I'd definitely prefer to err on the side of being explicit so that it's immediately obvious that the fields are present in the data structure, and are empty in this instance, but may not always be empty.

Thanks for your feedback here!

noamnelke commented on common/types/activation.go:
```diff
@@ -32,9 +33,9 @@ func (l EpochID) Field() log.Field { return log.Uint64("epoch_id", uint64(l)) }
 func (t ATXID) ShortString() string {
-	return t.Hash32().ShortString()
+	return t.Hash32().MediumString()
 }
```
I find this confusing... We should be consistent about terminology. If we want to return a medium, 10-character string (which I don't understand the reasoning for) then let's call the method `MediumString()`... Calling it `ShortString()` and then returning 10 characters is inconsistent.

noamnelke replied:
Thinking about this some more, I think that having 2 representations is bad. We should decide if we want 5 or 10 characters and then only use that and call it `ShortString()`.

I can't justify having two printed formats (other than full / short).

noamnelke commented on common/types/block.go:
```diff
@@ -11,16 +11,21 @@ import (
 
+// AtxsPerBlockLimit indicates the maximum number of atxs a block can reference
+const AtxsPerBlockLimit = 100
+
```
Why move this out of where the logic is (the `miner` package) and here, where the types are defined?

The previous location is more intuitive to me.

noamnelke commented on common/types/block.go:
```diff
@@ -152,6 +154,21 @@ func (b *Block) Bytes() []byte {
+// Fields returns an array of LoggableFields for logging
+func (b *Block) Fields() []log.LoggableField {
+	return []log.LoggableField{
+		log.BlockID(b.ID().String()),
+		log.LayerID(uint64(b.LayerIndex)),
```
```suggestion
		b.ID(),
		b.LayerIndex,
```

noamnelke commented on common/types/block.go:
```diff
@@ -152,6 +154,21 @@ func (b *Block) Bytes() []byte {
+	return []log.LoggableField{
+		log.BlockID(b.ID().String()),
+		log.LayerID(uint64(b.LayerIndex)),
+		log.MinerID(b.MinerID().ShortString()),
```
`b.MinerID()` returns `*signing.PublicKey`. If we make that type a loggable field it would simplify this line.

Another option, if you want to keep the `miner_id` label, is to add a `.Field(label string)` method to the type and then call it from here with the desired label.

noamnelke commented on common/types/hashes.go:
```diff
@@ -171,10 +171,22 @@ func (h Hash32) String() string {
+}
+
+// Stringify shortens a string to a specified length
+func Stringify(s string, maxlen int) string {
```
This is great, but maybe `Shorten()` is a better name?

noamnelke commented on config.toml:
```diff
@@ -1,6 +1,6 @@
 # sample spacemesh config file
 # use the config flag to start a node with a config file.
-# e.g $./go-spacemash --config ./config.toml
+# e.g $./go-spacemesh --config ./config.toml
```
changes to this file will conflict with your other PR...

lrettig replied:
appreciate the heads up! I've already rebased on the latest develop branch so I think we are good here!

noamnelke reviewed (commented):
Here's another batch of comments, more to come...

noamnelke commented on hare/eligibility/oracle.go:
```diff
@@ -185,19 +185,20 @@ func (o *Oracle) activeSetSize(layer types.LayerID) (uint32, error) {
-			log.String("id", id.ShortString()), log.Uint64("layer_id", uint64(layer)))
+		o.With().Info("eligibility: a node did not pass VRF signature verification",
+			log.String("id", id.ShortString()),
+			log.Uint64("layer_id", uint64(layer)))
```
both `id` and `layer` should be `LoggableField`s, iirc

noamnelke commented on hare/eligibility/oracle.go:
```diff
@@ -218,9 +219,11 @@ func (o *Oracle) Eligible(layer types.LayerID, round int32, committeeSize int, i
-			log.String("id", id.ShortString()), log.Int("committee_size", committeeSize),
-			log.Uint32("active_set_size", activeSetSize), log.Int32("round", round),
+		o.With().Info("eligibility: node did not pass VRF eligibility threshold",
+			log.String("id", id.ShortString()),
```
no need to wrap this, `id` should be a `LoggableField`.

noamnelke commented on log/zap.go:
```diff
@@ -100,39 +100,54 @@ func Duration(name string, val time.Duration) Field {
+func JobID(val interface{}) Field {
+	return String("job_id", fmt.Sprintf("%v", val))
+}
+
```
I rather not add more of these functions, they're sort of deprecated in favor of implementing the `LoggableField` interface on these types.

If the same type should often be logged with a different key, I think it's better to implement a `Field(key string)` method on the type than a method like these.

noamnelke commented on miner/block_builder.go:
```diff
@@ -29,9 +29,6 @@ const defaultFee = 1
 const IncomingTxProtocol = "TxGossip"
 
-// AtxsPerBlockLimit indicates the maximum number of atxs a block can reference
-const AtxsPerBlockLimit = 100
```
Maybe let's create a `constants` package and concentrate all of these there?

I don't think the `types` package is right for these.

Perhaps we should think about this some more - what should be in the config vs. a code constant? I think we should organize our config anyway, separating consensus/network parameters from "local" ones.

lrettig replied:
I can't remember why I moved this. I think it was a circular dependency error, but either I'm imagining that now or else it's gone away. I'll double-check, but for now I've put it back.

Feel free to open a new issue to follow up on creating a `constants` package if you think this is something we should do! It's definitely beyond the scope of this PR.

noamnelke commented on miner/block_builder.go:
```diff
@@ -290,14 +289,15 @@ func (t *BlockBuilder) createBlock(id types.LayerID, atxID types.ATXID, eligibil
 		log.Int("view_edges", len(bl.ViewEdges)),
 		log.Int("vote_count", len(bl.BlockVotes)),
-		bl.ATXID,
+		log.AtxID(bl.ATXID.ShortString()),
```
accident?

noamnelke commented on oracle/blockoracle.go:
```diff
@@ -73,7 +75,11 @@ func (bo *MinerBlockOracle) BlockEligible(layerID types.LayerID) (types.ATXID, [
-	bo.log.Info("miner %v found eligible for %d blocks in layer %d", bo.nodeID.Key[:5], len(proofs), layerID)
+	bo.log.With().Info("eligible for blocks in layer",
+		log.NodeID(bo.nodeID.ShortString()),
+		layerID.Field(),
```
```suggestion
		bo.nodeID,
		layerID,
```

noamnelke commented on oracle/blockoracle.go:
```diff
@@ -124,8 +130,30 @@ func (bo *MinerBlockOracle) calcEligibilityProofs(epochNumber types.EpochID) err
-		bo.nodeID.Key[:5], numberOfEligibleBlocks, len(bo.eligibilityProofs), epochNumber)
+
+	// Sort the layer map so we can print the layer data in order
+	keys := []types.LayerID{}
```
IDE recommends declaring this like: `var keys []types.LayerID`.

Here's further reading: https://stackoverflow.com/a/29164565/1225467

lrettig replied:
my lack of golang knowledge is showing 😊 good to know, thanks! and thanks for the link.

noamnelke commented on oracle/blockoracle.go:
```diff
@@ -124,8 +130,30 @@ func (bo *MinerBlockOracle) calcEligibilityProofs(epochNumber types.EpochID) err
+	})
+
+	// Pretty-print the number of blocks per eligible layer
+	strs := []string{}
```
Nil slice declaration via literal.

noamnelke commented on oracle/blockoracle.go:
```diff
@@ -124,8 +130,30 @@ func (bo *MinerBlockOracle) calcEligibilityProofs(epochNumber types.EpochID) err
+	strs := []string{}
+	for k := range keys {
+		strs = append(strs, fmt.Sprintf("Layer %d: %d", uint64(k), keys[k]))
+	}
```
You've got this wrong...

The `k` here is the index of the layer ID in the slice (an enumeration), while `keys[k]` is the actual layer ID.

```suggestion
	for k := range keys {
		strs = append(strs, fmt.Sprintf("Layer %d: %d", keys[k], len(bo.eligibilityProofs[k])))
	}
```

noamnelke commented on oracle/blockoracle.go:
```diff
@@ -124,8 +130,30 @@ func (bo *MinerBlockOracle) calcEligibilityProofs(epochNumber types.EpochID) err
+	// Sort the layer map so we can print the layer data in order
+	keys := []types.LayerID{}
+	for k := range bo.eligibilityProofs {
+		if len(bo.eligibilityProofs[k]) > 0 {
```
This should never be zero. I'd say better safe than sorry, but you're printing the length below anyway, so I think this check can be removed.

noamnelke commented on oracle/blockoracle.go:
```diff
@@ -124,8 +130,30 @@ func (bo *MinerBlockOracle) calcEligibilityProofs(epochNumber types.EpochID) err
+
+	bo.log.With().Info("eligibility for blocks in epoch",
+		log.NodeID(bo.nodeID.ShortString()),
+		log.EpochID(uint64(epochNumber)),
```
please use loggable fields

noamnelke commented on p2p/gossip/protocol.go:
```diff
@@ -186,15 +196,19 @@ func (prot *Protocol) handlePQ() {
+		prot.Log.With().Debug("new_gossip_message_relay",
+			log.String("from", m.Sender().String()),
+			log.String("protocol", m.Protocol()),
+			log.String("hash", util.Bytes2Hex(h[:])))
```
```suggestion
			h.Field("hash")
```

noamnelke commented on p2p/server/msgserver.go:
```diff
@@ -217,7 +217,10 @@ func (p *MessageServer) SendRequest(msgType MessageType, payload []byte, address
+		p.With().Error("sending message failed",
+			log.Uint32("msg_type", uint32(msgType)),
+			log.String("recipient", address.String()),
+			log.Int("msglen", len(payload)), log.Err(sendErr))
```
```suggestion
			log.Int("msglen", len(payload)),
			log.Err(sendErr))
```

noamnelke commented on signing/signer.go:
```diff
@@ -22,7 +22,11 @@ func NewPublicKey(pub []byte) *PublicKey {
 // Bytes returns the public key as byte array
 func (p *PublicKey) Bytes() []byte {
-	return p.pub
+	// Prevent segfault if unset
+	if p != nil {
+		return p.pub
+	}
+	return []byte{}
```
As noted in a previous comment, a `nil` go slice is equivalent to an empty slice. This change should be undone.

lrettig replied:
Please look at this again. The problem here isn't if `p.pub` is nil. The problem is if `p` itself is nil, which causes a segfault. I changed the last line here to return `nil` instead.

noamnelke replied:
Hmm, I see... I really misread that. I guess it's fine now, then.

noamnelke commented on sync/syncer.go:
```diff
@@ -329,7 +329,9 @@ func (s *Syncer) synchronise() {
 func (s *Syncer) handleWeaklySynced() {
-	s.With().Info("Node is weakly synced ", s.LatestLayer(), s.GetCurrentLayer())
+	s.With().Info("Node is weakly synced",
+		s.LatestLayer().Field(),
+		s.GetCurrentLayer().Field())
```
```suggestion
		s.LatestLayer(),
		s.GetCurrentLayer())
```

noamnelke commented on sync/syncer.go:
```diff
@@ -539,7 +541,7 @@ func (s *Syncer) syncLayer(layerID types.LayerID, blockIds []types.BlockID) ([]*
 		return nil, fmt.Errorf("failed adding layer %v blocks to queue %v", layerID, err)
 	} else if res == false {
-		s.With().Info("no missing blocks for layer", log.LayerID(layerID.Uint64()))
+		s.With().Info("no missing blocks for layer", layerID.Field())
```
```suggestion
		s.With().Info("no missing blocks for layer", layerID)
```

noamnelke commented on sync/syncer.go:
```diff
@@ -641,7 +643,9 @@ func (s *Syncer) validateBlockView(blk *types.Block) bool {
 	} else if res == false {
-		s.With().Info("block has no missing blocks in view", log.BlockID(blk.ID().String()), log.LayerID(uint64(blk.LayerIndex)))
+		s.With().Debug("no missing blocks in view",
+			log.BlockID(blk.ID().String()),
+			log.LayerID(uint64(blk.LayerIndex)))
```
use loggable fields

noamnelke commented on sync/validation_queue.go:
```diff
@@ -85,51 +85,52 @@ func (vq *blockQueue) handleBlocks(bjb fetchJob) {
 func (vq *blockQueue) handleBlock(id types.Hash32, block *types.Block) {
-	vq.Info("start handle block %s miner id %s", block.ID(), block.MinerID().ShortString())
+	vq.With().Info("start handling",
+		log.BlockID(block.ID().String()), log.MinerID(block.MinerID().ShortString()))
```
loggable fields

noamnelke commented on sync/validation_queue.go:
```diff
@@ -85,51 +85,52 @@ func (vq *blockQueue) handleBlocks(bjb fetchJob) {
 		return
 	}
-	vq.Info("finish fast validation block %v", block.ID())
+	vq.With().Info("finish fast validation", log.BlockID(block.ID().String()))
```
loggable fields

noamnelke commented on sync/validation_queue.go:
```diff
@@ -85,51 +85,52 @@ func (vq *blockQueue) handleBlocks(bjb fetchJob) {
 // if there are unknown blocks in the view they are added to the fetch queue
 func (vq *blockQueue) handleBlockDependencies(blk *types.Block) {
-	vq.Debug("handle dependencies Block %v", blk.ID())
+	vq.With().Debug("handle dependencies", log.BlockID(blk.ID().String()))
```
loggable fields

noamnelke commented on sync/validation_queue.go:
```diff
@@ -85,51 +85,52 @@ func (vq *blockQueue) handleBlocks(bjb fetchJob) {
 	if err != nil {
 		vq.updateDependencies(blk.Hash32(), false)
-		vq.Error(fmt.Sprintf("failed to add pending for Block %v %v", blk.ID(), err))
+		vq.With().Error("failed to add dependencies", log.BlockID(blk.ID().String()), log.Err(err))
```
loggable fields

noamnelke replied:
ok, not commenting about this anymore, please do it throughout the PR.

noamnelke commented on sync/validation_queue.go:
```diff
@@ -208,28 +209,32 @@ func (vq *blockQueue) addDependencies(jobID interface{}, blks []types.BlockID, f
 			//	check database
 			if _, err := vq.GetBlock(id); err != nil {
-				//unknown block add to queue
+				//add unknown block to queue
```
I think we should always add a space after the `//` this is inconsistent throughout this code, up to you if you want to fix (if you tell your IDE you want a space there, it's just a matter of using its format tool = one click).

lrettig replied:
the scope of this PR is fixing logs, not comments 😊 will fix the ones I see!

noamnelke reviewed (changes requested):
Almost all of my comments are minor, but there was one major comment where you log the wrong thing.

lrettig committed:
remove MediumString, set ShortString to len 10

per Noam feedback

lrettig committed:
Stringify -> Shorten

lrettig committed:
Take advantage of LoggableFields throughout

Still WIP. Per Noam feedback.
Make PublicKey loggable.

lrettig committed:
Update oracle/blockoracle.go

Co-authored-by: Noam Nelke <NoamNelke@gmail.com>

lrettig committed:
Update oracle/blockoracle.go

Co-authored-by: Noam Nelke <NoamNelke@gmail.com>

lrettig committed:
Update p2p/gossip/protocol.go

Co-authored-by: Noam Nelke <NoamNelke@gmail.com>

lrettig committed:
Update p2p/server/msgserver.go

Co-authored-by: Noam Nelke <NoamNelke@gmail.com>

lrettig committed:
Update sync/syncer.go

Co-authored-by: Noam Nelke <NoamNelke@gmail.com>

lrettig committed:
Update sync/syncer.go

Co-authored-by: Noam Nelke <NoamNelke@gmail.com>

lrettig committed:
Clean up code that prints eligibility per layer

Include all layers, per @noamnelke
Avoid append, since we know the length of the array
Use LoggableFields

lrettig committed:
nil go slice is equivalent to an empty slice

per @noamnelke

lrettig committed:
LoggableFields

per @noamnelke review
also, add space before some comments

lrettig committed:
Remove unnecessary log Field methods

Per @noamnelke feedback. Implement Field directly on type where
possible.

lrettig committed:
Move AtxsPerBlockLimit back to original location

And fix a syntax error typo

lrettig committed:
Some final cleanup

Use LoggableFields in a few more places
Some whitespace improvements

noamnelke reviewed (approved):
fix broken tests

lrettig committed:
Adjust ATXId len in test

Should be 10, not 12

lrettig commented:
Found the circular dependency issue. `common/types/block_test.go` relies on `miner.AtxsPerBlockLimit` but the miner package relies on `common.types`. This is why I moved `AtxsPerBlockLimit` into `common.types` before. I need to sort this out. Maybe this belongs in a separate `constants` package as @noamnelke suggested. Or maybe I can just refactor this test.

```
> go test -timeout 0 -p 1 -count 1 -v github.com/spacemeshos/go-spacemesh/common/types -run TestFields
# github.com/spacemeshos/go-spacemesh/common/types
import cycle not allowed in test
package github.com/spacemeshos/go-spacemesh/common/types (test)
        imports github.com/spacemeshos/go-spacemesh/miner
        imports github.com/spacemeshos/go-spacemesh/activation
        imports github.com/spacemeshos/go-spacemesh/common/types
FAIL    github.com/spacemeshos/go-spacemesh/common/types [setup failed]
FAIL
```

lrettig committed:
Clean up test

Doesn't rely on `AtxsPerBlockLimit` anymore, fixing circular dependency
issue - see
https://github.com/spacemeshos/go-spacemesh/pull/1952#issuecomment-637749731

lrettig committed:
Replace nil with empty string

Handle commitmentStr and challenge the same way

lrettig commented:
bors merge
